### PR TITLE
Support huggingface_hub 0.5

### PR DESCRIPTION
**Pull request by lhoestq**
Following https://github.com/huggingface/datasets/issues/4105

`huggingface_hub` deprecated some parameters in `HfApi` in 0.5. This PR updates all the calls to HfApi to remove all the deprecations, <s>and I set the `hugginface_hub` requirement to `>=0.5.0`</s>

cc @adrinjalali @LysandreJik

### CODE REVIEW

**Comment by lhoestq** [comment]
Looks like GH actions is not able to resolve `huggingface_hub` 0.5.0, I'm investivating

**Comment by adrinjalali** [comment]
I'm glad to see changes in `huggingface_hub` are simplifying code here.

**Comment by julien-c** [comment]
seems to supersede #4102, feel free to close mine :)

**Comment by albertvillanova** [comment]
I found the issue:
- https://github.com/huggingface/huggingface_hub/pull/790

**Comment by lhoestq** [comment]
Good catch, `huggingface_hub` doesn't support python 3.6 anymore indeed, therefore we should keep support for 0.4.0. I'm reverting the requirement version bump for now.

We can update the requirement once we drop support for python 3.6 in `datasets`

**Comment by albertvillanova** [comment]
@lhoestq, I've opened this PR on `huggingface_hub`: 
- https://github.com/huggingface/huggingface_hub/pull/823

Is there a strong reason wht `huggingface_hub` no longer support Python 3.6?

**Comment by lhoestq** [comment]
I think `datasets` can drop support for 3.6 soon. But for now maybe let's keep support for 0.4.0, python 3.6 users are not affected by https://github.com/huggingface/datasets/issues/4105 anyway.

`huggingface_hub` doesn't not have to support 3.6 again just for the CI IMO

**Comment by albertvillanova** [comment]
@lhoestq I commented on the PR, that IMO it is not a good practice to drop support for Python 3.6 without a previous deprecation cycle.

**Comment by lhoestq** [comment]
Re-added support for older versions. I ended up checking `huggingface_hub` version to use the old, deprecated API for <0.5.0

**Comment by adrinjalali** [comment]
I find it good practice to have all dependency version related code in a single file so that when you decide to remove support for an old version of a dependency it's easy to find and remove them, hence suggesting `utils/_fixes.py` in https://github.com/huggingface/datasets/issues/4105#issuecomment-1090041204

**Comment by lhoestq** [comment]
good idea, thanks !

**Comment by lhoestq** [comment]
I used your suggestion @adrinjalali , I just replace the try/except with a check on the version of `huggingface_hub`

File: tests/test_upstream_hub.py [hunk 1]
```diff
@@ -25,34 +27,25 @@
...
 @with_staging_testing
 class TestPushToHub(TestCase):
     _api = HfApi(endpoint=ENDPOINT_STAGING)
 
+    def cleanup_repo(self, ds_name):
+        if version.parse(huggingface_hub.__version__) < version.parse("0.5.0"):
+            organization, name = ds_name.split("/")
+            self._api.delete_repo(name, token=self._token, repo_type="dataset", organization=organization)
+        else:  # the `organization` parameter is deprecated in huggingface_hub>=0.5.0
+            self._api.delete_repo(ds_name, token=self._token, repo_type="dataset")
```
Comment (adrinjalali) [inline tests/test_upstream_hub.py:47]: same with this one?

Comment (lhoestq) [reply tests/test_upstream_hub.py]: done ! thanks

**Review by adrinjalali — APPROVED** [review]

**Review by albertvillanova — APPROVED** [review]
Thanks for the fix.

**Pull request merged**